### ISSUE TITLE
SubscribeSingleInvoice returns no error and no invoice data on nonexistent hash

### ISSUE THREAD

**Issue by akovalenko** [lightningnetwork/lnd#6424]
### Background

I consider using SubscribeSingleInvoice instead of polling for hodl invoice states. There's a problem, almost a show-stopper: when I request subscription to nonexistent payment hash, there's no error and no initial stream message. I understand the logic here, as the invoice with this payment hash may appear later by AddHoldInvoice, but...

Canceled invoices are eventually deleted, so any service which analyzes invoice states and which might have some long downtime should be prepared to handle missing invoice as Canceled. That's what I cannot do here without unreliable tricks (like "if it doesn't respond with the initial state in 5 seconds, then it's canceled"  or "let's do lookupInvoice first and pray that it isn't deleted between lookupInvoice and SubscribeSingleInvoice").

I'd like some guidance on how I should monitor invoices reliably.  

### Your environment

* version of `lnd`: current git master 0b20dcc4f52341a3aa3e9f245daecc8497feea32
* which operating system (`uname -a` on *Nix) `Linux murphy 5.13.0-30-generic #33~20.04.1-Ubuntu SMP Mon Feb 7 14:25:10 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`
* version of `btcd`, `bitcoind`, or other backend: bitcoind 22.0

### Steps to reproduce

SubscribeSingleInvoice for nonexistent hash

### Expected behaviour

Instant NotFound error 

### Actual behaviour

invoiceStream not returning any initial state, just hanging forever

**Comment by alexbosworth** [comment]
Could you look up the invoice first before subscribing to its state updates?

**Comment by akovalenko** [comment]
I can and I'll probably have to. But if an old canceled invoice could be removed automatically between lookup and subscription, I get unexpecting hanging subscription again.

I've also noticed that subscription to unknown hash hangs forever, even if such a hash materialize later through AddHoldInvoice, there are no events. It suggests that the hanging behavior is unintentional (I had some doubts), and replacing it with error report would probably hurt no one.

**Comment by alexbosworth** [comment]
Yeah if it never works that seems like there is no reason not to cancel the subscription, either it should work eventually or it should fail immediately

**Comment by akovalenko** [comment]
Looking at invoices/invoiceregistry.go: 
```go
        // It is possible that the invoice does not exist yet, but the client is
        // already watching it in anticipation.
        if err == channeldb.ErrInvoiceNotFound ||
                err == channeldb.ErrNoInvoicesCreated {

                return nil
        }
```
So the intent was to allow tracking not-yet-appeared hashes. I'll recheck to be sure it doesn't work this way (I may be wrong because my testing environment was messy), but anyway it is no good when canceled invoices may be removed.
